dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/executor/logging/LocalLogClient.java

```diff
@@ -0,0 +1,91 @@
+     *
+     * @param taskInstance The task instance object, containing information needed to retrieve the log.
+     * @return The complete log file download response of the task instance, including log content and metadata.
+     */
+    public TaskInstanceLogFileDownloadResponse getWholeLog(TaskInstance taskInstance) {
+        return getLocalWholeLog(taskInstance);
+    }
+
+    /**
+     * Query a portion of the log of a task instance.
+     * This method is used to query log information of a task instance in a paginated manner,
+     * suitable for scenarios where the log content is large and needs to be retrieved in batches.
+     *
+     * @param taskInstance The task instance object, containing information needed to retrieve the log.
+     * @param skipLineNum The number of lines to skip, indicating from which line to start reading the log.
+     * @param limit The maximum number of lines to read, indicating the maximum number of lines to retrieve in this query.
+     * @return The partial log query response, including log content within the specified range and metadata.
+     */
+    public TaskInstanceLogPageQueryResponse getPartLog(TaskInstance taskInstance, int skipLineNum, int limit) {
+        return getLocalPartLog(taskInstance, skipLineNum, limit);
+    }
+
+    private static TaskInstanceLogFileDownloadResponse getLocalWholeLog(TaskInstance taskInstance) {
+        TaskInstanceLogFileDownloadRequest request = buildLogFileDownloadRequest(taskInstance);
+        return getProxyLogService(taskInstance).getTaskInstanceWholeLogFileBytes(request);
+    }
+
+    private static TaskInstanceLogPageQueryResponse getLocalPartLog(TaskInstance taskInstance, int skipLineNum,
+                                                                    int limit) {
+        TaskInstanceLogPageQueryRequest request = buildLogPageQueryRequest(taskInstance, skipLineNum, limit);
+        return getProxyLogService(taskInstance).pageQueryTaskInstanceLog(request);
+    }
+
+    private static TaskInstanceLogFileDownloadRequest buildLogFileDownloadRequest(TaskInstance taskInstance) {
+        return new TaskInstanceLogFileDownloadRequest(
+                taskInstance.getId(),
+                taskInstance.getLogPath());
+    }
+
+    private static TaskInstanceLogPageQueryRequest buildLogPageQueryRequest(TaskInstance taskInstance, int skipLineNum,
+                                                                            int limit) {
+        return TaskInstanceLogPageQueryRequest.builder()
+                .taskInstanceId(taskInstance.getId())
+                .taskInstanceLogAbsolutePath(taskInstance.getLogPath())
+                .skipLineNum(skipLineNum)
+                .limit(limit)
+                .build();
+    }
+
+    private static ILogService getProxyLogService(TaskInstance taskInstance) {
+        return Clients
+                .withService(ILogService.class)
+                .withHost(taskInstance.getHost());
+    }
```

Comment:
Please don't use static here, and the method `buildLogFileDownloadRequest`, `buildLogFileDownloadRequest` seems meaningless, this is only use once.